storage/src/main/java/org/apache/kafka/server/log/internals/FetchParams.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.log.internals;
+
+import org.apache.kafka.common.replica.ClientMetadata;
+import org.apache.kafka.common.utils.FetchRequestUtils;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class FetchParams {
+    private final short requestVersion;
+    private final int replicaId;
+    private final long maxWaitMs;
+    private final int minBytes;
+    private final int maxBytes;
+    private final FetchIsolation isolation;
```

Comment:
We can make these fields public since they're immutable. The accessors don't add much value in such cases.